src/drivers/pca9685_pwm_out/PCA9685.h

```diff
@@ -91,7 +91,12 @@ namespace drv_pca9685_pwm
 #define PCA9685_CLOCK_FREQ PCA9685_CLOCK_EXT   // use ext clk
 #endif
 
+#define PCA9685_DEVICE_BASE_PATH	"/dev/pca9685"
 #define PWM_DEFAULT_FREQUENCY 50    // default pwm frequency
+/* Correct for overshoot in the frequency setting (see issue
+ * https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library/issues/11).
+ */
+#define PCA9685_FREQ_OVERSHOOT_FACTOR 0.9f
```

Comment:
This is not the real fix.
https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library/issues/40
https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library/pull/61
https://github.com/hybridgroup/gobot/pull/582/files

Reply:
> This is not the real fix.
> [adafruit/Adafruit-PWM-Servo-Driver-Library#40](https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library/issues/40)
> [adafruit/Adafruit-PWM-Servo-Driver-Library#61](https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library/pull/61)

Reverted. Now it already uses `round()` logic. Should we introduce the measurement of the actual frequency that osc oscillated at? That is not user-friendly, even developer as well.
I think a macro which simply decides the osc freq at compile time would almost solve the problem as it is too hard to get the actual freq.
I have tested through a few PCA9685 chips and all of them seems have an osc higher than 25MHz, just like the condition that above links discussed.

Reply:
8231762